tasks/Module 4/task_2.py

```diff
@@ -0,0 +1,31 @@
+"""Fill in the Gaps."""
+import os
+
+
+def fill_gaps(folder_path, prefix):
+    """Fill in the Gaps."""
+    # init file list with first not needed element
+    file_list = [None]
+    os.chdir(folder_path)
+    # get list of tuples of filenames and extentions
+    with os.scandir(folder_path) as scandir:
+        for entry in scandir:
+            if entry.name.startswith(prefix) and entry.is_file():
+                file_list.append(os.path.splitext(entry.name))
```

Comment:
Please consider appending here full filenames and splitting them into base name and extension in the next loop. IMHO it may be more readable, especially that you still need original filename there.

Reply:
I am not sure i got an idea with full filenames. What for i would need it ?

Reply:
I meant that you would need original filename to rename the file: `os.renames(''.join(file_data), new_file_name)`. If you keep original full filename in the list you can do just `os.renames(filename, new_file_name)`. It is not a big deal however, so do what you like more.